### PR TITLE
chore: fix npm-publish dependencies and add provenance

### PR DESCRIPTION
**Pull request by legendecas**
The `npm-publish` steps depends on output variable `release-please.outputs.release_created`. Without this variable the steps are skipped, e.g. https://github.com/nodejs/node-gyp/actions/runs/12087642317.

---

Adds npm provenance statements: https://docs.npmjs.com/generating-provenance-statements#example-github-actions-workflow.

---

The release-please bot created PR needs an additional token to trigger CI runs: 
>    When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.
>    https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow#triggering-a-workflow-from-a-workflow

Removing the always skipped release tests for now.

### CODE REVIEW

**Comment by legendecas** [comment]
@nodejs/node-gyp would you mind taking a look at this so that we can release v10.3.0? Thank you!

File: .github/workflows/tests.yml
Comment (cclauss) [inline .github/workflows/tests.yml:22]: This assumption does not make me comfortable. When you vendor-in source code, it becomes your code, and you should run your own tests on it.

Comment (lukekarrys) [reply .github/workflows/tests.yml:22]: Can we run two separate checks? It would be great to surface issues in `/gyp` because we do vendor it. But in the past it has created friction because a linting error in `/gyp` code would fail all of CI.

File: .github/workflows/release-please.yml
Comment (lukekarrys) [reply .github/workflows/release-please.yml:7]: I see now why this was added. The `10.3.0` release was created (https://github.com/nodejs/node-gyp/releases/tag/v10.3.0) but never published. So this would allow us to publish from CI (and with `--provenance`) after these cases.

`node-gyp` currently does not have separate permissions for write access to the repo vs merging a release PR. If it did, then we would not want to add `workflow_dispatch` as a trigger because it would elevate anyone with write access to being able to publish a release.

Since we don't currently do this, I'm ok with adding this workflow_dispatch trigger since it is currently more important to be able to publish 😄

File: .github/workflows/tests.yml
Comment (lukekarrys) [reply .github/workflows/tests.yml:22]: @legendecas Can we revert this part and re-open as a new PR? It would be nice to unblock the release from CI and figure out python linting in a separate discussion.

**Review by lukekarrys — APPROVED** [review]
Approved but I think the `ruff check` changes should move to a separate PR discussion.

File: .github/workflows/release-please.yml
Comment (legendecas) [reply .github/workflows/release-please.yml:7]: I added this to allow triggered release-please manually. I think this doesn't expose publishing directly, as the `npm-publish` step still requires the release-please to be run first and actually "creates a releasee".

Comment (lukekarrys) [reply .github/workflows/release-please.yml:7]: That makes sense now and I don't think I was reading the workflow correctly.

> Taking a step back, I think this could be removed since the release-please is triggered on every push. So formally we should land PRs to re-trigger release.

Here is the issue I've seen in the past with release-please workflows:

1. The release workflow goes bad (for any reason) and is left in an inconsistent state. Currently this repo is in that state because we have a GitHub release for `10.3.0` but no npm publish for that versions
1. Make new commits to fix what went wrong with the release
2. `release-please` picks up those new commits and makes `10.3.1`
3. If it all works then `10.3.1` is correctly released and published everywhere

But this is not the ideal outcome because `10.3.0` was never published. So there needs to be some way to `npm publish`. That could be via `workflow_dispatch` or someone needs to do it manually.

File: .github/workflows/tests.yml
Comment (lukekarrys) [reply .github/workflows/tests.yml:22]: Thanks for the explanation @legendecas, I didn't realized CI was failing. I think this change should be landed to make CI green and a new issue/PR opened to get two different `ruff` checks running, each with the correct configuration.

File: .github/workflows/release-please.yml
Comment (legendecas) [reply .github/workflows/release-please.yml:7]: Actually, I think it could be "fine-ish" to just go ahead and publish `10.3.1` to make this easier 😮‍💨

Comment (lukekarrys) [reply .github/workflows/release-please.yml:7]: I agree (for various definitions of "fine-ish" 😄). In that case, I think `workflow_dispatch` won't be necessary to include in this PR?

But I do think it's an ok outcome to merge this and get CI fixed and have 10.3.1 published.

Since I have publish rights from my personal npm account, I would like to publish https://github.com/nodejs/node-gyp/commit/9afaf00e7ab547fd76c6890582abff9739effa24 from my local machine as `10.3.0` to keep a clean paper trail 🧹

Comment (legendecas) [reply .github/workflows/release-please.yml]: yeah, removed.

**Review by lukekarrys — APPROVED** [review]

**Pull request merged**
